File: 프로그래머스/1/87389. 나머지가 1이 되는 수 찾기/나머지가 1이 되는 수 찾기.java
class Solution {
    public int solution(int n) {
        
        int answer = findNum(n-1);
        
        return answer;
    }
    
    public int findNum(int num){
        for(int i=2; i<=Math.sqrt(num); i++){
            if(num%i == 0) return i;
        }
        return num;
    }
}

File: 프로그래머스/1/87389. 나머지가 1이 되는 수 찾기/나머지가 1이 되는 수 찾기.js
function solution(n) {
    
    var answer = find(n-1);
    
    return answer;
}

function find(n){
    
    for(let i=2; i<=Math.sqrt(n); i++){
        if(n%i == 0) return i;
    }
    
    return n;
        
}